src/DataValues/TimeValueCalculator.php

```diff
@@ -21,6 +21,22 @@ class TimeValueCalculator {
 	 */
 	const SECONDS_PER_GREGORIAN_YEAR = 31556952;
 
+	/**
+	 * Maximum length for a timestamp.
+	 */
+	private $MAX_LENGTH_TIMESTAMP = 33;
+
+	/**
+	 * Lowest positive timestamp.
+	 */
+	private $TIMESTAMP_ZERO = '+0000000000000000-01-01T00:00:00Z';
+
+	/**
+	 * Highest positive timestamp strictly earlier than the lowest positive timestamp with
+	 * a length of $MAX_LENGTH_TIMESTAMP + 1.
+	 */
+	private $HIGHEST_TIMESTAMP = '+9999999999999999-12-31T23:59:59Z';
```

Comment:
I still feel this needs to be 61 seconds, even if I'm not sure if this is relevant for anything, because wikidata.org currently does not allow precisions smaller than a day anyway.

Reply:
:+1: